File: Java/219-ContainsDuplicateII.java
public class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Set<Integer> hs = new HashSet<>();
        for (int i = 0; i < nums.length; i ++){
            if (i > k)
                hs.remove(nums[i - k - 1]);
            if (!hs.add(nums[i]))   
                return true;
        }
        return false;
    }
}

File: cpp/219-ContainsDuplicateII.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_set<int> hs;
        for (int i = 0; i < nums.size(); i++) {
            if (hs.size() > k) {
                //attention!!! i - k - 1 !
                hs.erase(nums[i-k-1]);
            }
            if (hs.find(nums[i]) != hs.end()) {
                return true;
            }
            hs.insert(nums[i]);
        }
        return false;
    }
};
